XLite support

bsongis opened the pull request:
[X10] Serial on external module
[X10] Many fixes
[ARM] Backlight check moved to readKeysAndTrims() on all ARM boards. So that the backlight is again ON when pressing a trim or the SHIFT key

bsongis committed:
[X10] Many fixes, [ARM] Backlight

[X10] Serial on external module (not tested)
[X10] Many fixes
[ARM] Backlight check moved to readKeysAndTrims() on all ARM boards. So that the backlight is again ON when pressing a trim or the SHIFT key

bsongis committed:
[X-Lite ] External module driver

bsongis committed:
Merge remote-tracking branch 'origin/2.2' into bsongis/xlite_fixes

# Conflicts:
#	radio/src/gui/128x64/radio_hardware.cpp

bsongis committed:
Compilation fix

bsongis committed:
Compilation fix

bsongis committed:
Compilation fix

bsongis committed:
[X-lite] S.PORT and BT fixes

bsongis committed:
Compilation fixes

bsongis committed:
Compilation fixes

bsongis committed:
Compilation fixes

bsongis committed:
[X-Lite] S.PORT now OK

bsongis committed:
3djc/xlite bt (#5815)

* Add Bt trainer in model menu. WARNING : lines are not disabled when hardware BT is not set to trainer

* Xlite : trainer lines disabled in model setup when BT is disabled in hardware

* Fixes

* cleanup

bsongis committed:
Compilation fix

bsongis committed:
Compilation fix

3djc committed:
Display only the two physical trims in key diags

3djc committed:
Better battery default

kilrah commented on radio/src/opentx.cpp:
```diff
@@ -285,8 +285,13 @@ void generalDefault()
+  // 2 x Li-Ion
+  g_eeGeneral.vBatWarn = 66;
+  g_eeGeneral.vBatMin = -23;  // 6.7V
+  g_eeGeneral.vBatMax = -37;  // 8.3V  
```
Hmm I'd change for X10 too, it's supposed to be the same after all...

3djc committed:
Move USEHORUSBT to taranis only

3djc committed:
Compile fix

bsongis committed:
Refactoring

3djc committed:
fix ext receiver flashing, tested ok (and tested x7 ok too)

3djc committed:
Fix leds colors (tested by FrSky on a fixed radio)

3djc committed:
Fix xlite logs

bsongis committed:
One more TODO

3djc committed:
One less TODO

bsongis committed:
[X-Lite] Names edition

I would prefer we use new #define, like NAVIGATION_9X etc.

bsongis committed:
[X-Lite] Names edition. For discussion ...

Better like this?

bsongis committed:
[X-Lite] No value change if not in EDIT mode

bsongis committed:
[X-Lite] Switches inverted

bsongis committed:
[X-Lite] Companion support continued

kilrah committed:
VC++ compilation

3djc committed:
GVARS default on
LUA compile fixes (needs further functional tests

3djc committed:
More LUA fixes

3djc committed:
SHIFT long does not exist, add xlite KEY_FIRST

3djc committed:
Proposal for LUA support of shift+long, needs to be discussed

bsongis commented on radio/src/keys.cpp:
```diff
@@ -106,7 +107,10 @@ void Key::input(bool val)
         // generate long key press
         // TRACE("key %d LONG", key());
-        putEvent(EVT_KEY_LONG(key()));
+#if !defined(LUA)
```
It shouldn't be dependent of LUA / NO LUA

bsongis commented on radio/src/keys.cpp:
```diff
@@ -96,7 +96,8 @@ void Key::input(bool val)
     case KSTATE_START:
       // TRACE("key %d FIRST", key());
-      putEvent(EVT_KEY_FIRST(key()));
+      if (!IS_SHIFT_KEY(key()))
```
Perhaps this test can be removed, I think the only problem we have with the SHIFT key is KEY_REPEAT

3djc committed:
Update 12x64 scripts for xlite compatibility

kilrah committed:
Batt calib, FR fixes

kilrah committed:
String length

kilrah committed:
Pot type defaults

3djc committed:
Only KEY_REPEAT needs to be disabled for SHIFT

3djc committed:
Typo

3djc committed:
Improve switch warning UI for 128x64 Taranis (#5822)

* - Match X9 behavior
- Made to handle replacing SH by a two pos and the future two additional switches
- Use string tables to find switch standard designation

* Compile fix

kilrah committed:
DBLKEYS

kilrah committed:
Fix pot type selection and calibration on 128x64 (menu settings were there but completely ignored!)

kilrah committed:
Taranis variants only

kilrah committed:
Companion fixes

kilrah committed:
We have 4 trims in the end

kilrah committed:
Companion general defaults

kilrah committed:
Source/switch popup on Xlite

kilrah committed:
Trim strings no more a special case for Xlite

kilrah committed:
Fixes the "autoscroll" issue but no idea why it isn't an issue on others...

kilrah committed:
Unnecessary test generated compiler warning

kilrah committed:
Reset fix

kilrah committed:
Merge branch '2.2' into bsongis/xlite_fixes

# Conflicts:
#	radio/src/gui/128x64/model_setup.cpp

kilrah commented:
Current status:
- Xlite reboots fine from bootloader exit and a watchdog reset.
- Xlite recovery from USB mass storage is still random.
- X9E does not reboot from either bootloader exit or watchdog reset, but that is the case on the 2.2 branch as well so it got broken earlier.
- Page key and trims are broken on X7.

bsongis commented on radio/src/gui/128x64/model_setup.cpp:
```diff
@@ -700,7 +729,7 @@ void menuModelSetup(event_t event)
         break;
       }
-#if defined(PCBTARANIS)
+  #if defined(PCBTARANIS)
```
indentation

bsongis commented on radio/src/gui/128x64/navigation.cpp:
```diff
@@ -344,7 +346,7 @@ int checkIncDec(event_t event, int val, int i_min, int i_max, unsigned int i_fla
   if (s_editMode > 0) {
 #endif
-    if (event==EVT_KEY_FIRST(KEY_RIGHT) || event==EVT_KEY_REPT(KEY_RIGHT) || (s_editMode>0 && (IS_ROTARY_RIGHT(event) || event==EVT_KEY_FIRST(KEY_UP) || event==EVT_KEY_REPT(KEY_UP)))) {
+    if (event==EVT_KEY_FIRST(KEY_RIGHT) || event==EVT_KEY_REPT(KEY_RIGHT)) {
```
you break SKY9X here

kilrah replied:
Well it disables editing with up/down for it too, but we want that for consistency no?

bsongis commented on radio/src/gui/128x64/navigation.cpp:
```diff
@@ -355,7 +357,7 @@ int checkIncDec(event_t event, int val, int i_min, int i_max, unsigned int i_fla
       }
     }
-    else if (event==EVT_KEY_FIRST(KEY_LEFT) || event==EVT_KEY_REPT(KEY_LEFT) || (s_editMode>0 && (IS_ROTARY_LEFT(event) || event==EVT_KEY_FIRST(KEY_DOWN) || event==EVT_KEY_REPT(KEY_DOWN)))) {
+    else if (event==EVT_KEY_FIRST(KEY_LEFT) || event==EVT_KEY_REPT(KEY_LEFT)) {
```
same problem here

kilrah committed:
Don't change sky9x nav

kilrah committed:
Typo

3djc committed:
Alternative nav

kilrah committed:
Restore module power init, cleaned up a bit too much

kilrah commented:
Shall we merge and finish the last things in 2.2?

3djc committed:
Nav issue

bsongis commented:
I have opened a new branch for next changes ...
